Several concurrent bug fixes related to shipping of AddonHeader changes.

KevinMind opened the pull request:
fixes mozilla/addons#15695
fixes mozilla/addons#15698
fixes mozilla/addons#15699
fixes mozilla/addons#15700
fixes mozilla/addons#15704
fixes mozilla/addons#15707
fixes mozilla/addons#15710
fixes mozilla/addons#15713

## Description

- remove number from users badge when no active users
- install button does not shrink in loading state
- add "Experimental" test on experimental badge
- reformat grid names and spacing to make it more clear the structure of the layout
- fixes broken styles on StaticAddonCard for use in the blog(s)
- make layout more consistent with figma

## Testing

### **Test Scenario 1: Experimental Badge Simplification**
**Focus**: Verify that experimental badges display correctly with icon and text

**Steps:**
1. Navigate to an add-on page with an experimental add-on (see ticket)
2. Look for the "Experimental" badge in the addon badges section
3. **Verify**: The experimental badge displays as a text + icon badge
4. **Verify**: The badge text reads "Experimental" 
5. **Verify**: The badge styling appears consistent with other badges

**Expected Result**: Experimental badge shows the badge icon plus the text

---

### **Test Scenario 2: User Count Badge Formatting**
**Focus**: Verify proper number formatting and pluralization in user count badges

**Test Cases:**

**2a. Zero Users**
1. Find an add-on with 0 average daily users (or use dev tools to mock this)
2. **Verify**: Badge displays "No Users"

**2b. Single User**  
1. Find an add-on with exactly 1 average daily user
2. **Verify**: Badge displays "1 User" (singular)

**2c. Multiple Users**
1. Find an add-on with multiple users (e.g., 100 users)
2. **Verify**: Badge displays "100 Users" (plural)

**2d. Large Numbers**
1. Find an add-on with thousands of users (e.g., 12,345 users)
2. **Verify**: Badge displays "12,345 Users" with proper comma formatting
3. **Verify**: Numbers are formatted according to locale (commas for thousands in English)

NOTE: you can set up this scenario easily on local dev by overriding the addon.average_daily_users property to any number.

---

### **Test Scenario 3: Addon Page Layout and Grid Structure**
**Focus**: Verify the new responsive grid layout works correctly

**Test Device Sizes:**
- Mobile (< 768px)
- Tablet (768px - 1024px)  
- Desktop (> 1024px)
- Large Desktop (> 1200px)

**Steps:**
1. Navigate to any add-on page
2. Resize browser window to test different breakpoints
3. **Verify**: Layout adapts properly at each breakpoint
5. **Verify**: Icon, info, badges, and install button are positioned correctly
6. **Verify**: No overlapping or misaligned elements
7. **Verify**: Grid areas (preview, icon, info, badges, button) flow logically

**Expected Result**: Responsive layout that works well on all screen sizes. Importantly the button is not on the bottom of the layout until we get to extraExtraLarge size. This causes fewer layout shifts in the design and is more consistent with the phase2 design.

---

### **Test Scenario 4: Warning Section Organization**  
**Focus**: Verify warnings are properly grouped and displayed

**Steps:**
1. Navigate to an add-on page that triggers warnings (try these scenarios):

**4a. Non-public Add-on**
1. View a disabled add-on or one with non-public status
2. **Verify**: Warning message appears in the warnings section
3. **Verify**: Message reads "This is not a public listing. You are only seeing it because of elevated permissions."

**4b. Install Error Warning**
1. Trigger an install error (may require dev tools or specific browser state)
2. **Verify**: Install error message appears in warnings section

**4c. Compatibility Warning**
1. View an add-on with platform compatibility issues
2. **Verify**: Compatibility error message appears in warnings section
3. **Verify**: Message indicates platform availability

**4d. General Install Warning**
1. View any regular add-on
2. **Verify**: General security warning appears
3. **Verify**: Message reads "This add-on is not actively monitored for security by Mozilla. Make sure you trust it before installing."

**4e. Wrong Platform Warning**
1. Access site from different platform context (may require user agent modification)
2. **Verify**: Platform-specific warning appears when applicable

**Expected Result**: All warnings are grouped together in a dedicated warnings section above the main addon information. This causes fewer layout shifts within the header as the warnings no longer disrupt the layout conditionally.

---

### **Test Scenario 5: Blog Utils Static Addon Card**
**Focus**: Verify simplified static addon card (for blog integration)

**Steps:**
1. If you have access to the blog integration or can test the static card component:
2. **Verify**: Rating stars are no longer displayed
3. **Verify**: User count metadata is no longer displayed  
4. **Verify**: Card shows only: icon, title, author, badges, summary, and Firefox download button
5. **Verify**: Layout is clean and simplified

**Expected Result**: Streamlined addon card without rating/user count information

NOTE: You can test this scenario locally following steps outlined in the repository.

1. run `yarn build:blog-utils-dev`
2. open `http://127.0.0.1:11000`
3. verify the addon card is rednering correctly

**Addon:**

<img width="657" alt="image" src="https://github.com/user-attachments/assets/569b4b83-bf51-43eb-81dd-1c34f24e7aa0" />

**Theme:**

<img width="673" alt="image" src="https://github.com/user-attachments/assets/f16daf76-a50c-45f6-8eb7-2bb96e64fdd4" />

---

### **Test Scenario 6: Cross-browser Compatibility**
**Focus**: Verify changes work across different browsers

**Browsers to Test:**
- Firefox (latest)
- Chrome (latest)  
- Safari (if available)
- Edge (latest)

**Steps:**
1. Repeat key scenarios (user count formatting, layout, warnings) in each browser
5. **Verify**: Consistent appearance and functionality
6. **Verify**: No browser-specific rendering issues
7. **Verify**: Grid layout works in all browsers

NOTE: for the most part you should test the breakpoint changes to the grid layout for both extensions and themes.

---

### **Test Scenario 7: Visual Regression Testing**
**Focus**: Ensure styling changes appear as intended

**Steps:**
1. Compare addon pages before and after changes
2. **Verify**: Badge styling is consistent and visually appealing  
3. **Verify**: Grid layout looks clean and professional
4. **Verify**: Warning section is clearly distinguished
5. **Verify**: No unintended visual artifacts or spacing issues

---

### **Test Scenario 8: Non-firefox browser**
**Focus**: Ensure card header is appropriately spaced

**Steps:**
1. **Verify**: The install button shows "get firefox" and no longer shows the yellow get firefox callout embedded tooltip
---

### **Test Environment Setup:**
- Use staging environment: `https://addons.allizom.org/`
- Test both logged-in and logged-out states
- Clear browser cache between tests
- Test with various add-on types (extensions, themes, etc.)

### **Critical Test Data Needed:**
- Add-ons with 0, 1, and many users for count testing
- Experimental add-ons for badge testing  
- Non-public add-ons for warning testing
- Add-ons with different statuses and compatibility issues

Each scenario should be tested systematically, and any deviations from expected behavior should be reported as bugs with specific details about browser, viewport size, and reproduction steps.

willdurand commented on src/amo/components/AddonBadges/index.js:
```diff
@@ -149,13 +143,14 @@ export class AddonBadgesBase extends React.Component<InternalProps> {
+        ? [
+            i18n.formatNumber(averageDailyUsers),
+            i18n.ngettext('User', 'Users', averageDailyUsers),
+          ].join(' ')
```
In general, you should not concatenate or join localized strings. The way to translate this string is this example: https://github.com/mozilla/addons-frontend/blob/17b5f91f196619cdf455c4ac19e3ae75b641a5c6/src/amo/components/UserCollection/index.js#L45-L48

willdurand replied:
Also, since the existing test coverage didn't catch this, we should improve the test in this PR.

KevinMind replied:
Ah great point. Yeah this is supposed to be a WIP Pr. I have not even touched tests.

willdurand commented on src/amo/pages/Addon/styles.scss:
```diff
@@ -37,67 +37,71 @@
 
   & .Addon-theme-thumbnail {
-    grid-area: preview;
+    grid-area: prv;
```
It'd be good to keep actual names instead of obscure 3-letter ids.

willdurand commented:
On Slack, you mentioned that this patch was also fixing https://github.com/mozilla/addons/issues/15700, is that correct? Please update the title of the PR accordingly (e.g. `More fixes to the add-on header`) and add some "Fixes #XYZ" for each issue being addressed here (`XYZ` being an issue number).

KevinMind commented:
Will do. I may actually split this to multiple PRs not totally sure yet. For sure will split to separate commits. But I'll do that tomorrow. I should have opened as draft cause it's not ready for review (though valid feedback already achieved)

KevinMind commented:
@willdurand I have
- added the missing fix for static card
- updated tests to cover the changes
- I haven't reorganized the commits.

Looking at it I think splitting to multiple PRs will increase effort for little beefit. Instead I propose to add detailed test instructions for each scenario so we can get it verified with the rest of stuff on dev tomorrow/monday.

I think we are still in the "possible" camp in terms of getting this merged and verified before the tag depending on how code review from here goes.

willdurand commented:
> fixes [mozilla/addons#15700](https://github.com/mozilla/addons/issues/15700)

Looks like the theme variant needs some more work still:

<img width="714" alt="Screenshot 2025-07-07 at 10 49 59" src="https://github.com/user-attachments/assets/7851b9a8-a23e-40ec-b114-5811d59942f3" />

It isn't clear to me why the badges tend to be right-aligned:

<img width="742" alt="Screenshot 2025-07-07 at 10 50 53" src="https://github.com/user-attachments/assets/1fe1ba45-f8f7-4219-ac8b-e8b862a75e2a" />

willdurand reviewed (changes requested):
#### In a non-Firefox browser, the install button is a bit too narrow at ~750px and the icons are all over the place.

<img width="833" alt="Screenshot 2025-07-07 at 11 16 12" src="https://github.com/user-attachments/assets/31139f22-0e7d-4580-9400-8c14f83f726a" />

#### In a non-Firefox browser, do we expect the install button to be kinda "above" the title? Leaving quite some empty space above said title?

<img width="1624" alt="Screenshot 2025-07-07 at 11 17 46" src="https://github.com/user-attachments/assets/8c13b956-924c-46d6-8851-306e6f7ad966" />

#### On a narrow screen, it looks like badges may overflow:

![127 0 0 1_3000_fr_firefox_addon_dsa-test-6_](https://github.com/user-attachments/assets/63beded4-cd46-4dfc-b9e0-6d8bf1d982d5)

![127 0 0 1_3000_fr_firefox_addon_dsa-test-6_ (1)](https://github.com/user-attachments/assets/d9fb0366-e9a1-431c-bb4b-404e555cf733)


This is prod (for comparison):

![addons mozilla org_fr_firefox_addon_emr-to-daspayments-integration_](https://github.com/user-attachments/assets/e2788584-e85b-43db-b4b1-1176c21b7346)

#### Install button and loading state

Question: When installing an add-on with the `mozAddonManager` (see [docs](https://addons-frontend.readthedocs.io/en/latest/moz-addon-manager/#developing-with-a-local-https-server-recommended) to get that working locally), is it expected to have the button in "loading" state aligned on one side? (as opposed to centered?)

https://github.com/user-attachments/assets/1621dc3e-d1b0-4c6a-a858-dc68aeb703e7

https://github.com/user-attachments/assets/e2ef919c-c6ea-45a4-8c06-50e5811ed2d6

willdurand commented:
> Question: When installing an add-on with the `mozAddonManager` (see [docs](https://addons-frontend.readthedocs.io/en/latest/moz-addon-manager/#developing-with-a-local-https-server-recommended) to get that working locally), is it expected to have the button in "loading" state aligned on one side? (as opposed to centered?)

I see https://github.com/mozilla/addons/issues/15698 was filed with more or less the same question.

KevinMind commented on src/amo/components/AddonBadges/index.js:
```diff
@@ -71,13 +71,7 @@ export class AddonBadgesBase extends React.Component<InternalProps> {
         size="large"
-      >
-        {(props) => (
-          <BadgePill {...props}>
```
When you do not pass an explicit render function it renders the content and the pill. this change is effectively "restoring" the text.

willdurand commented on src/amo/components/GetFirefoxButton/index.js:
```diff
@@ -154,18 +154,6 @@ export const GetFirefoxButtonBase = ({
   const buttonText = supportsRTAMO
     ? downloadTextForRTAMO
     : i18n.gettext('Download Firefox');
-  let calloutText =
```
Where is the decision captured for that? And why is it not a different PR?

willdurand reviewed (approved):
Looks ok, though I have a question about the info text when the install button is shown on non-fx browsers.
